Holovibes/sources/API.cc

```diff
@@ -1236,72 +1143,44 @@ std::vector<float> get_input_filter() { return GET_SETTING(InputFilter); }
 
 void set_input_filter(std::vector<float> value) { UPDATE_SETTING(InputFilter, value); }
 
-void load_input_filter(std::vector<float> input_filter, const std::string& file)
+void load_input_filter(const std::string& file)
 {
     auto& holo = Holovibes::instance();
     try
     {
         auto path_file = dir / __INPUT_FILTER_FOLDER_PATH__ / file;
-        InputFilter(input_filter,
+        InputFilter(get_input_filter(),
                     path_file.string(),
                     holo.get_gpu_output_queue()->get_fd().width,
                     holo.get_gpu_output_queue()->get_fd().height);
     }
     catch (std::exception& e)
     {
-        api::set_input_filter({});
         LOG_ERROR("Couldn't load input filter : {}", e.what());
     }
-    pipe_refresh();
 }
 
 void enable_filter(const std::string& filename)
 {
-    UserInterfaceDescriptor::instance().filter_name = filename;
-    enable_filter();
-}
-
-void enable_filter()
-{
-    auto filename = UserInterfaceDescriptor::instance().filter_name;
-    auto file = filename == UID_FILTER_TYPE_DEFAULT ? std::nullopt : std::make_optional(filename);
+    if (filename == api::get_filter_file_name())
+        return;
 
+    api::set_filter_file_name(filename);
     UPDATE_SETTING(FilterEnabled, true);
-    set_input_filter({});
-
-    if (get_compute_pipe_no_throw() != nullptr)
-    {
-        // There is no file None.txt for filtering
-        if (file && file.value() != UID_FILTER_TYPE_DEFAULT)
-            load_input_filter(get_input_filter(), file.value());
-        else
-            disable_filter();
-
-        // Refresh because the current filter might have change.
-        // pipe_refresh();
 
-        if (filename == UID_FILTER_TYPE_DEFAULT)
-            return;
-
-        pipe_refresh();
-    }
-}
+    if (!get_compute_pipe_no_throw())
+        return;
 
-void disable_filter()
-{
-    set_input_filter({});
-    UPDATE_SETTING(FilterEnabled, false);
-    try
-    {
-        auto pipe = get_compute_pipe();
-        pipe->request(ICS::DisableFilter);
-        while (pipe->is_requested(ICS::DisableFilter))
-            continue;
-    }
-    catch (const std::exception& e)
+    // There is no file for filtering
+    if (filename.empty())
     {
-        LOG_ERROR("Catch {}", e.what());
+        set_input_filter({});
+        UPDATE_SETTING(FilterEnabled, false);
     }
```

Comment:
(For the 'green' part)
Maybe move the checks before updating any setting ? You also could use variables and ternary operators to only call UPDATE_SETTING and set_input_filter once i guess, idk